include/broker/zeek.hh

```diff
@@ -105,6 +130,104 @@ public:
     return get<std::string>(get<vector>(as_vector()[2])[0]);
   }
 
+  /// Support iteration with structured binding.
+  class metadata_iterator {
```

Comment:
1) This header is using Zeek's coding style, so: `MetadataIterator`.
2) Having this type inline introduces a tight coupling between the types that isn't really necessary. Moving this to namespace scope IMHO makes the individual parts better to read.